src/AmplNLWriter.jl

```diff
@@ -47,13 +48,19 @@ function call_solver(
     stdout::IO,
 )
     solver.f() do solver_path
-        ret = run(
-            pipeline(
+        # Solvers like Ipopt_jll use libblastrampoline. That requires us to set
+        # the BLAS library via the LBT_DEFAULT_LIBS environment variable.
+        # Provide a default in case the user doesn't set.
+        blas = OpenBLAS32_jll.libopenblas
+        solver_cmd = pipeline(
+            addenv(
                 `$(solver_path) $(nl_filename) -AMPL $(options)`,
-                stdin = stdin,
-                stdout = stdout,
-            ),
+                "LBT_DEFAULT_LIBS" => get(ENV, "LBT_DEFAULT_LIBS", blas),
```

Comment:
If you want to get _really_ fancy, you can auto-include the LP64 libraries that are currently loaded into Julia's copy of LBT:

```
using OpenBLAS32_jll
using LinearAlgebra

BLAS.lbt_forward(OpenBLAS32_jll.libopenblas_path)
```

```
default_libs = join([lib.libname for lib in BLAS.get_config().loaded_libs if lib.interface == :lp64], ";")
```